src/components/PageSizeSelector/PageSizeSelector.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import Select from "components/Select";
+import { observer, inject } from "mobx-react";
+
+const PAGE_SIZES = [
+  {
+    name: "20 Products",
+    value: 20
+  },
+  {
+    name: "60 Products",
+    value: 60
+  },
+  {
+    name: "100 Products",
+    value: 100
+  }
+];
+
+@inject("uiStore")
+@observer
+class PageSizeSelector extends Component {
+  static propTypes = {
+    uiStore: PropTypes.object.isRequired
+  }
+
+  handleChange = (event) => {
+    this.props.uiStore.setPageSize(event.target.value);
+  }
```

Comment:
I would make this `handleChange` the same as the one in `Select`, calling `this.props.onChange`. Then call `setPageSize` in `ProductGrid` instead.

My reason is because we may want to copy/import the page size selector elsewhere. Ideally we will move it to our component library for others to use, and they might not use mobx.

So the props for this component should be `onChange` and `pageSize`, and no uiStore prop or injection.

Reply:
I have elevated event handling to the page level and removed mobx dependencies from `PageSizeSelector`